app/models/resource.rb

```diff
@@ -20,6 +21,19 @@ class Resource < ApplicationRecord
 
   validates :title, :resource_type, presence: true
 
-  after_save { SearchIndex.add(self) }
-  after_destroy { SearchIndex.remove(self) }
+  after_save do
+    SearchIndex.new(
+      model_name: self.class.name,
+      id: id,
+      async: true,
+    ).add
+  end
+
+  after_destroy do
+    SearchIndex.new(
```

Comment:
> Would it make sense to have RemoveFromIndexJob directly here instead of having the path go `SearchIndex` -> `RemoveFromIndexJob` -> `SearchIndex`?

Yeah I debated this while writing it and settled on the `async: true` flag to hide the job stuff away. However, since you raised it, I'm going to call this "confusing" and just create the new job directly from `Resource` a new PR.

> Alternatively, would it make sense to have the code to actually add or remove the item from the index to exist within AddToIndexJob and RemoveFromIndexJob?

This is what I was hoping to avoid with `SearchIndex` -  abstract as much of the ElasticSearch-specific stuff in one utility class as I could. That's actually why I set up the runaround described above.